### PR TITLE
refactor(load3d): extract Viewport3d base + SceneOverlay protocol

### PR DESCRIPTION
**Pull request by jtydhr88**
## Summary
Split Load3d into Viewport3d (model-agnostic viewport scaffolding) and Load3d (extends Viewport3d, adds loader/model/animation/HDRI/recording/gizmo). 
Viewport3d exposes only render-loop plumbing, layout, mouse status, camera orchestration, and the SceneOverlay protocol so future 3D node viewports can compose it without inheriting model machinery.

- SceneOverlay protocol (attach/detach/update/onActiveCameraChange/dispose) gives any 3D node a managed lifecycle slot for plugging scene content into the viewport.
- Viewport3d.setExternalActiveCamera(cam | null) for POV swap: renders from an externally-owned camera (e.g. a subject camera authored by an overlay), with OrbitControls detached and the view helper hidden.
- ControlsManager.detach()/attach() back the POV control gating.
- Two-phase init via Viewport3d.start() so subclass field assignments finish before any render-path code dispatches through overridden tickPerFrame / isActive.

Behavior preserved for all 5 Load3d consumers (Load3D, Preview3D, PreviewGaussianSplat, PreviewPointCloud, SaveGLB). 
Sets up the upcoming CreateCameraInfo preview and other future Three.js-based 3D nodes (Pose Editor, Animation Director) to compose Viewport3d plus their own SceneOverlay implementation.

### CODE REVIEW

File: src/extensions/core/load3d/Load3d.ts [hunk 1]
```diff
@@ -553,19 +311,15 @@ class Load3d {
...
 
-  /**
-   * Monotonic counter that ticks once per loadModel call, **before** any
-   * await. Callers can capture this immediately after triggering a load and
-   * later compare against `currentLoadGeneration` to verify their load is
-   * still the latest one — useful when chaining post-load work
-   * (e.g. applying camera matrices) through `whenLoadIdle()`, which would
-   * otherwise wait for any newer queued load and apply stale state to it.
-   */
   get currentLoadGeneration(): number {
```
Comment (dante01yoon) [inline src/extensions/core/load3d/Load3d.ts:323]: nit-pick (non-blocking): this getter's docstring (explaining the monotonic load-generation counter and why it guards against applying stale camera state through `whenLoadIdle()`) was dropped here, and the `setupForModel runs during loadModel and clobbers the camera; restore on top.` comment was dropped near the `savedCameraState` restore (~L386). Those are outside the Viewport3d extraction and explain genuinely non-obvious *why*, which the repo guidance values keeping. Could we retain those two? Fine to leave the `// fill container` / `// Animation methods` what-comments removed.

**Review by dante01yoon — APPROVED** [review]
Approving. This is a faithful, behavior-preserving extraction of `Viewport3d` out of `Load3d`.

What I verified:
- **Render/tick order preserved for Load3d**: `tickPerFrame` override runs `animationManager.update(delta)` then `super.tickPerFrame` (overlay → viewHelper → controls), matching the old inline order; `renderMainScene` renders `getRenderCamera()` which falls back to `cameraManager.activeCamera` (Load3d never sets an external camera), so the rendered camera is unchanged.
- **toggleCamera**: Load3d no longer overrides it — it routes the gizmo update through the new `onActiveCameraChanged()` hook and lets the base notify the overlay. This both preserves the old gizmo/controls/viewHelper/resize ordering and resolves CodeRabbit's earlier "override loses overlay notification" comment.
- **Two-phase init via `start()`**: render-path dispatch (`handleResize`/`forceRender`/`startAnimation`) is deferred until after subclass fields are assigned. The base constructor doesn't dispatch to any overridden method; the `ResizeObserver` and context-menu callbacks are async/user-driven, so `Load3d.tickPerFrame` can't run before `animationManager` exists. `INITIAL_RENDER_DONE` is still set synchronously by end of construction (via `start()`→`handleResize`→`forceRender`).
- **Lifecycle fix**: `remove()` now clears `initialRenderTimer`, closing a latent post-dispose `forceRender` against a disposed renderer.
- **Consumers**: all 5 node types go through `createLoad3d` → `new Load3d` → constructor `this.start()`; no `Load3d`/`Viewport3d` subclasses exist, so no external `start()` call is needed.
- **Tests**: new unit tests are behavioral (overlay lifecycle, POV swap, `applyTargetSize` guards, timer cleanup) and not change-detectors; the existing load3d Playwright suite (green, 1669 passed) covers the WebGL render/resize plumbing.

Non-blocking:
- All 4 CodeRabbit inline comments target the earlier commit `ca57fe0b` and are resolved on the current head; their `CHANGES_REQUESTED` state is stale (their latest review reports no actionable comments).
- `codecov/patch` (57% < 76%) is informational here — the uncovered lines are the render/resize plumbing that the e2e suite exercises. The Vercel check is an infra rate-limit.

One small nit inline.
